insights/client/apps/compliance/__init__.py

```diff
@@ -69,8 +69,9 @@ def download_tailoring_file(self, policy):
     # TODO: Not a typo! This endpoint gives OSCAP policies, not profiles
     # We need to update compliance-backend to fix this
     def get_policies(self):
+        search = 'system_names={0} canonical=false os_minor_version={1}'.format(self.hostname, self.os_minor_version())
```

Comment:
> If get_policies is to gather policies that the system is assigned to, then it should not filter them by an OS version.

Before, we were looking for the initial profile (`external = false`). Swapping that for os_minor_version still gives us unique 'policies' (used very loosely, any one profile in each policy). This query gives exact tab matches. It's true that this PR does not yet consider when the tab does not exist.

> external=false should be kept to retrieve policy assignment (w/o any OS filtering). Otherwise there can be a confusion when the OS is updated (and/or not configured on a policy). In that case user will receive info on that the system isn't assigned, which in reality it can be. This feature should be only around tailoring and not policy assignment, unless I've missed something.

100% agree with this. Sorry, I didn't make it clear that this PR is only half-baked. Here's what I was thinking:

1. Find policy profiles that match exactly by tab (i.e. os_minor_version)
2. Find initial policy profiles (the old API request with external=false)
3. Scan based on (ref_ids+tailorings from 1) + (ref_ids from 2 without tailorings - ref_ids from 1)

Reply:
I'd reorder the steps to have the "Find initial policy profiles" first, unless you wanted to have it as a fallback. What you are suggesting makes sense, as it starts using the correct objects. It kinda feels that it moves the selection logic to the client, and also considering that it does so by using filters.

My proposal is/was to keep the semantics with just the initial policy profile, by forcing `tailored=true` and passing OS info on the tailoring endpoint. The idea is that client would request assigned policies, and for each policy it would request tailoring. Tailoring selection would be done on backend with the information provided by client. (policy and initial profile is interchangeable)